core/error_macros.h

```diff
@@ -470,25 +473,19 @@ void _err_print_index_error(const char *p_function, const char *p_file, int p_li
  *
  * The current function returns.
  */
-#define ERR_FAIL()                                                                     \
-	if (1) {                                                                           \
-		_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Method/Function Failed."); \
-		return;                                                                        \
-	} else                                                                             \
-		((void)0)
+#define ERR_FAIL()                                                                 \
+	_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Method/function failed."); \
+	return
```

Comment:
Dropping the `if` wrapper will break functions that have their own `if (error)` checks without braces: `{}` i.e. it will only print the error, but it won't return. The `if` wrapper is needed to allow multi-line replacements for one line statements.

Reply:
Ah I see. And just replacing if by an extra block would prevent using the `((void)0)` hack indeed. So I guess I'll revert those changes.